Sources/QuadTreeNode.swift

```diff
@@ -0,0 +1,83 @@
+import MapKit
+
+// MARK: - Tree node
+
+final class QuadTreeNode {
+
+  private let capacity: Int
+  private let rect: MKMapRect
+  private var children: [QuadTreeNode] = []
+  private var annotations: [MKAnnotation] = []
+
+  var isLeaf: Bool {
+    return children.isEmpty
+  }
+
+  init(rect: MKMapRect, capacity: Int) {
+    self.rect = rect
+    self.capacity = capacity
+  }
+
+  @discardableResult
+  func add(annotation: MKAnnotation) -> Bool {
+    guard rect.contains(annotation: annotation) else {
+      return false
+    }
+
+    if isLeaf {
+      annotations.append(annotation)
+      // Subdivide if needed
+      if annotations.count == capacity {
```

Comment:
`capacity` indicates how many annotations each node has. When `annotations.count == capacity` a node will subdivide itself and become a `branch` node with 4 children. Then every new annotation will go to the child node if its rect contains a given annotation map point.